src/test/run-pass/binding/fn-arg-incomplete-pattern-drop-order.rs

```diff
@@ -0,0 +1,54 @@
+// Check that partially moved from function parameters are dropped after the
+// named bindings that move from them.
+
+use std::{panic, cell::RefCell};
+
+struct LogDrop<'a>(i32, Context<'a>);
+
+#[derive(Copy, Clone)]
+struct Context<'a> {
+    panic_on: i32,
+    drops: &'a RefCell<Vec<i32>>,
+}
+
+impl<'a> Context<'a> {
+    fn record_drop(self, index: i32) {
+        self.drops.borrow_mut().push(index);
+        if index == self.panic_on {
+            panic!();
+        }
+    }
+}
+
+impl<'a> Drop for LogDrop<'a> {
+    fn drop(&mut self) {
+        self.1.record_drop(self.0);
+    }
+}
+
+fn foo((_x, _): (LogDrop, LogDrop), (_, _y): (LogDrop, LogDrop)) {}
```

Comment:
Could this test also have the `let` version, to explicitly demonstrate that they do the same thing?

Like
```rust
fn foo2(_temp1: (LogDrop, LogDrop), _temp2: (LogDrop, LogDrop)) {
    let (_x, _) = _temp1;
    let (_, _y) = _temp2;
}
```